### ISSUE TITLE
Expired RRSIG DNSSEC keys not being regenerated/refreshed/extended

### ISSUE THREAD

**Issue by bjmuld** [mail-in-a-box/mailinabox#1940]
Nightly backup routine recently started sending notification of unresolving DNS (remote backup machine is on same domain).  Looking into the issue reveals expired RRSIG records.  

I found some old issues related to this and so I tried the following to no avail:
```
sudo ~/mailinabox/management/dns_update.py --force
sudo ~/mailinabox/tools/dns_update --force
sudo systemctl restart bind9
sudo systemctl restart nsd
```

**Comment by jvolkenant** [comment]
I don't believe there is a `--force` argument for `management/dns_update.py`, even still I don't think the logic will force update the RRSIG keys until 3 days or less of them expiring.  I'm not sure the history behind this and if we want to force update it so often.  I can't imagine it would hurt anything but would require a PR that updates it when forced

https://github.com/mail-in-a-box/mailinabox/blob/6653dbb2e2357462d22685a06f688f1fb02ed3cd/management/dns_update.py#L521-L535

**Comment by bjmuld** [comment]
Well, in either case, MIAB still hasn't updated my RRSIG keys, so I have disabled DNSSEC with my registrar.  I'm just not sure how to troubleshoot this any further...

**Comment by jvolkenant** [comment]
Check that there are any errors with the mail in a box daemon `grep "start\[" /var/log/syslog`, You might want to restart the daemon as well `systemctl restart mailinabox`. Also what do these tools show for your domain?

https://mxtoolbox.com/rrsig.aspx
https://dnsviz.net/

**Comment by bjmuld** [comment]
nothing in syslog
both mxtoolbox and dnsviz show expired RRSIG keys

**Comment by myfirstnameispaul** [comment]
Does anyone recall what is the period of expiration for the keys? Mine are set to expire 20210414 and my current installation was the first release on 18.04, whatever version number whenever that was.

**Comment by jvolkenant** [comment]
30 days?
https://github.com/mail-in-a-box/mailinabox/blob/6653dbb2e2357462d22685a06f688f1fb02ed3cd/management/dns_update.py#L664

**Comment by bjmuld** [comment]
Any ideas why `dns_update.py` did _not_ refresh expired DNSSEC signatures for me?  I have had DNSSEC turned off for the last two weeks or so, and don't want to continue like this...

**Comment by ddavness** [comment]
I know I'm kind of late to the party, but did you try by any chance adding a dummy TXT record and then removing it? May be of help, because that will force the daemon to rebuild the whole zone

**Comment by bjmuld** [comment]
Turns out, I had a redirect entry in `custom.yaml` that was prompting creation of several DNS records (A,MX,TXT, _dmarc) while also having a CNAME entry that I manually created for that domain.  NSD was choking on the zone file because of the conflicting A and CNAME records, but remained semi-functional.  It would resolve portions of that domain successfully.  It was in troubleshooting the one subdomain that was mysteriously NOT resolving that led me to the problem.
